rfd/0163-vnet.md

```diff
@@ -426,39 +426,51 @@ nameserver 100.127.100.127
 The first version of VNet will only support MacOS so this is written with that
 in mind, but we can implement this in a similar way for Windows and Linux.
 
-#### Admin process
+#### Daemon
 
-All system configuration must run as a privileged user (root).
+All system configuration must run as a privileged user (root). To achieve that, VNet starts a daemon
+with `tsh vnet daemon` that runs as root, a Launch Daemon in macOS terms. The daemon handles
+creating a TUN device (and passing back its file descriptor or [GUID on
+Windows](https://github.com/WireGuard/wireguard-go/blob/12269c2761734b15625017d8565745096325392f/tun/tun_windows.go#L66-L68)).
 
-In the demo implementation we are using an Applescript wrapper to call a child
-process as root.
-Alternatives are considered under the
-[Alternatives to osascript](#alternatives-to-osascript) section.
-The privileged child process will still be the `tsh` binary, just called with
-specific arguments to do the host setup.
+On macOS, the daemon is added through [`SMAppService`](https://developer.apple.com/documentation/servicemanagement/smappservice?language=objc)
+and a plist that lives inside the app bundle. This has two nice properties. First, the daemon and
+the plist are signed by us, preventing 3rd parties from making any changes. Second, since the daemon
+plist lives in the app bundle, it is automatically removed when the user removes tsh.app or Teleport
+Connect.app.
 
-The admin process will be started in the background by `tsh vnet` when it first
-starts, or by Connect when VNet is first turned on (usually when it first opens
-after the first time VNet is enabled).
-It will watch or poll the contents of the `TELEPORT_HOME` directory for
-currently active Teleport profiles.
+`SMAppService` requires macOS 13.0+. Since it only works with signed packages, in development we can
+substitute it with an osascript wrapper that runs the same command as root.
```

Comment:
How much would introducing codesigning change our current development/deployment/installation process for Connect? Is it worth it in a v1 if we could get away without SMAppService by creating a plist file for a launchd launch daemon  https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html?

Reply:
As discussed yesterday, we already do codesigning and we will eventually use SMAppService.

We could use `launchctl` directly, but now that I wrote a PoC of SMAppService, it doesn't really save us any time and introduces additional problems. For example, SMAppService automatically cleans up the daemon when trashing `.app` while with launchctl the user would have to manually remember to remove the daemon plist ([AFAIK](https://forums.developer.apple.com/forums/thread/66821?answerId=245716022#245716022)).